dojo/importers/default_importer.py

```diff
@@ -0,0 +1,472 @@
+            # Process any endpoints on the endpoint, or added on the form
+            self.process_endpoints(finding, kwargs.get("endpoints_to_add", []))
+            # Process any tags
+            if finding.unsaved_tags:
+                finding.tags = finding.unsaved_tags
+            # Process any files
+            self.process_files(finding)
+            # Process vulnerability IDs
+            finding = self.process_vulnerability_ids(finding)
+            # Categorize this finding as a new one
+            new_findings.append(finding)
+            # to avoid pushing a finding group multiple times, we push those outside of the loop
+            push_to_jira = kwargs.get("push_to_jira", False)
+            if is_finding_groups_enabled() and group_by:
+                finding.save()
+            else:
+                finding.save(push_to_jira=push_to_jira)
+
+        for (group_name, findings) in group_names_to_findings_dict.items():
+            finding_helper.add_findings_to_auto_group(
+                group_name,
+                findings,
+                **kwargs
+            )
+            if push_to_jira:
+                if findings[0].finding_group is not None:
+                    jira_helper.push_to_jira(findings[0].finding_group)
+                else:
+                    jira_helper.push_to_jira(findings[0])
+
+        sync = kwargs.get('sync', False)
+        if not sync:
+            return [serialize('json', [finding, ]) for finding in new_findings]
+        return new_findings
+
+    def close_old_findings(
+        self,
+        test: Test,
+        findings: List[Finding],
+        user: Dojo_User,
+        scan_date: datetime = timezone.now(),
+        **kwargs: dict,
+    ) -> List[Finding]:
+        """
+        Closes old findings based on a hash code match at either the product
+        or the engagement scope. Closing an old finding entails setting the
+        finding to mitigated status, setting all endpoint statuses to mitigated,
+        as well as leaving a not on the finding indicating that it was mitigated
+        because the vulnerability is no longer present in the submitted scan report.
+        """
+        # First check if close old findings is desired
+        if kwargs.get("close_old_findings") is False:
+            return []
+        logger.debug("REIMPORT_SCAN: Closing findings no longer present in scan report")
+        # Close old active findings that are not reported by this scan.
+        # Refactoring this to only call test.finding_set.values() once.
+        mitigated_hash_codes = []
+        new_hash_codes = []
+        for finding in findings:
+            new_hash_codes.append(finding["hash_code"])
+            if getattr(finding, "is_mitigated", None):
+                mitigated_hash_codes.append(finding["hash_code"])
+                for hash_code in new_hash_codes:
+                    if hash_code == finding["hash_code"]:
+                        new_hash_codes.remove(hash_code)
```

Comment:
Small tweak to avoid looping through all of `new_hash_codes` to remove every `hash_code` for mitigated findings. If necessary for the `exclude()` call, could just `list()` it at the end.

```suggestion
        new_hash_codes = set() 
        for finding in findings:
            if getattr(finding, "is_mitigated", None):
                mitigated_hash_codes.append(finding["hash_code"])
            else:
                new_hash_codes.add(finding["hash_code"])
```

Reply:
Punting for now to avoid potential breakages